### ISSUE TITLE
Ruff reports S604 on creating a shapely.geometry.Polygon

### ISSUE THREAD

**Issue by peterbandi** [astral-sh/ruff#27432]
### Summary

The following line in a pytest test file (e.g. _tests/test_something.py_) triggers **S604 Function call with truthy \`shell\` parameter identified, security issue"**, while it has nothing to do with `subprocess`:

```python
import shapely.geometry
geometry = shapely.geometry.Polygon(shell=[(0, 0), (3, 0), (3, 3), (0, 3), (0, 0)])
```

**The code only triggers the wrong S604 issue when in a pytest file.** Additionally, there are two workarounds (apart from disabling the rule):

```python
# Instantiating without the `shell` keyword does not trigger S604.
geometry_0 = shapely.geometry.Polygon([(0, 0), (3, 0), (3, 3), (0, 3), (0, 0)])

# Instantiating without a literal does not trigger S604 either.
polygon_shell = [(0, 0), (3, 0), (3, 3), (0, 3), (0, 0)]
geometry_1 = shapely.geometry.Polygon(shell=polygon_shell)
```

I used `shapely` 2.1.2 (_shapely=2.1.2=py312h383787d_2_ conda package from the conda-forge channel).

### Version

ruff 0.16.1 (_ruff=0.16.1=h462bb3b_0_ conda package from the conda-forge channel)

**Comment by ntBre** [comment]
Yeah, this is unfortunately the [known problem](https://docs.astral.sh/ruff/rules/call-with-shell-equals-true/#known-problems) called out in the docs:

> Prone to false positives as it is triggered on any function call with a `shell=True` parameter.

So this is intentional for parity with bandit and because this is a security rule. The [per-file-ignores](https://docs.astral.sh/ruff/settings/#lint_per-file-ignores) setting may be helpful if you want to disable it for all of your tests.

**Comment by peterbandi** [comment]
All right, thank you!
